addons/hr_expense/tests/test_expenses.py

```diff
@@ -1179,3 +1181,86 @@ def test_expense_from_attachments(self):
+
+    def test_expense_sheet_attachments_sync(self):
+        """
+        Test that the hr.expense.sheet attachments stay in sync with the attachments associated with the expense lines
+        Syncing should happen when:
+        - When adding/removing expense_line_ids on a hr.expense.sheet <-> changing sheet_id on an expense
+        - When deleting an expense that is associated with an hr.expense.sheet
+        - When adding/removing an attachment of an expense that is associated with an hr.expense.sheet
+        """
+        def assert_attachments_are_synced(attachments_from_expenses, attachments_from_sheet, attachments_on_sheet, sheet_has_attachment):
+            if sheet_has_attachment:
+                self.assertTrue(bool(attachments_on_sheet), "Attachment that belongs to the hr.expense.sheet only was removed unexpectedly")
+            self.assertSetEqual(set(attachments_from_expenses.mapped('checksum')), set((attachments_from_sheet - attachments_on_sheet).mapped('checksum')),
+                                "Attachments between expenses and their sheet is not in sync.")
+
+        for sheet_has_attachment in (False, True):
+            expense_with_attachment_1, expense_with_attachment_2, expense_with_attachment_3 = self.env['hr.expense'].create([{
+                'name': 'expense_1',
+                'employee_id': self.expense_employee.id,
+                'product_id': self.product_a.id,
+                'unit_amount': 1000.00,
+            }, {
+                'name': 'expense_2',
+                'employee_id': self.expense_employee.id,
+                'product_id': self.product_a.id,
+                'unit_amount': 999.00,
+            }, {
+                'name': 'expense_3',
+                'employee_id': self.expense_employee.id,
+                'product_id': self.product_a.id,
+                'unit_amount': 998.00,
+            }])
+            self.env['ir.attachment'].create([{
+                'name': "test_file_1.png",
+                'datas': base64.b64encode(b'content'),
+                'res_id': expense_with_attachment_1.id,
+                'res_model': 'hr.expense',
+            }, {
+                'name': "test_file_2.png",
+                'datas': base64.b64encode(b'other content'),
+                'res_id': expense_with_attachment_2.id,
+                'res_model': 'hr.expense',
+            }, {
+                'name': "test_file_3.png",
+                'datas': base64.b64encode(b'different content'),
+                'res_id': expense_with_attachment_3.id,
+                'res_model': 'hr.expense',
+            }])
+
+            expense_sheet = self.env['hr.expense.sheet'].create({
+                'company_id': self.env.company.id,
+                'employee_id': self.expense_employee.id,
+                'name': 'test sheet',
+                'expense_line_ids': [Command.set([expense_with_attachment_1.id, expense_with_attachment_2.id, expense_with_attachment_3.id])],
+            })
+
```

Comment:
To avoid re-writing your tests at every forward-ports, in your expenses creation use 
```
'product_id': self.product_c.id,
'total_amount': X,
```
instead of using product_a and overriding the unit_amount (and using a default quantity of 1)

Reply:
Thanks for the info and avoiding me potentially useless conflicts :)